semgrep/semgrep/core_runner.py

```diff
@@ -463,15 +461,9 @@ def handle_regex_patterns(
         except re.error as err:
             raise SemgrepError(f"invalid regular expression specified: {err}")
 
-        if self._testing:
-            # Testing functionality runs in a multiprocessing.Pool. We cannot run
-            # a Pool inside a Pool, so we have to avoid multiprocessing when testing.
-            # https://stackoverflow.com/questions/6974695/python-process-pool-non-daemonic
-            matches = [get_re_matches(patterns_re, target) for target in targets]
-        else:
-            re_fn = functools.partial(get_re_matches, patterns_re)
-            with multiprocessing.Pool(self._jobs) as pool:
-                matches = pool.map(re_fn, targets)
+        re_fn = functools.partial(get_re_matches, patterns_re)
+        with pool.ThreadPool(self._jobs) as tpool:
```

Comment:
This was originally a suggestion by @nbrahms. It allows us to more cleanly avoid this problem: https://stackoverflow.com/questions/6974695/python-process-pool-non-daemonic :+1: